### PR TITLE
feat(clone): snapshot allocated bytes to be updated based on parent snapshot for clone

### PR DESCRIPTION
**Pull request by hrudaya21**

### CODE REVIEW

File: io-engine/src/core/logical_volume.rs [hunk 1]
```diff
@@ -31,6 +32,9 @@ pub trait LogicalVolume {
     /// Return the committed size of the Logical Volume in bytes.
     fn committed(&self) -> u64;
 
+    /// Get BlobPtr from spdk_lvol.
```
Comment (tiagolobocastro) [inline io-engine/src/core/logical_volume.rs:35]:
```suggestion
    /// Get BlobPtr.
```

This is fine for now but I think in the future we'll need to refactor this as we're bringing already the second spdk specific method to an interface which is meant to be generic.

**Review by tiagolobocastro — APPROVED** [review]

File: io-engine/src/bin/io-engine-client/v1/replica_cli.rs [hunk 2]
```diff
@@ -331,6 +336,8 @@ async fn replica_list(
                     "URI",
                     "IS_SNAPSHOT",
                     "IS_CLONE",
+                    "SNAP_ANCSTR_SIZE",
```
Comment (Abhinandan-Purkait) [inline io-engine/src/bin/io-engine-client/v1/replica_cli.rs]: `SNAP_ANCESTOR_SIZE` Since only 2 letters maybe we can add?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: io-engine/src/bin/io-engine-client/v1/replica_cli.rs (hunk 2 above)
Comment (hrudaya21) [reply io-engine/src/bin/io-engine-client/v1/replica_cli.rs]: It was just kept to save space in the console to add new fields. :)
Anyways, I have changed it. In future if any new fields comes, then We will realing all the fields accordingly.

**Comment by hrudaya21** [comment]
bors try

**Comment by hrudaya21** [comment]
bors try

**Comment by hrudaya21** [comment]
bors merge

**Comment by hrudaya21** [comment]
bors merge

**Pull request merged**